src/Database/DriverInterface.php

```diff
@@ -180,9 +180,9 @@ public function queryTranslator(string $type): Closure;
      * If all the tables that use this Driver specify their
      * own schemas, then this may return null.
      *
-     * @return \Cake\Database\Schema\BaseSchema
+     * @return \Cake\Database\Schema\SchemaDialect
      */
-    public function schemaDialect(): BaseSchema;
+    public function schemaDialect(): SchemaDialect;
```

Comment:
Change return type would be a non BC change even of a new minor.

Reply:
There is `class_alias('Cake\Database\Schema\SchemaDialect', 'Cake\Database\Schema\BaseSchema');` so it shouldn't be so BC.

3.x had same with `Response` etc

Reply:
Oops, sorry I overlooked the fact that it's just a rename.